sdk/vaa/governance.go

```diff
@@ -23,3 +26,31 @@ func CreateGovernanceVAA(timestamp time.Time, nonce uint32, sequence uint64, gua
 
 	return vaa
 }
+
+// Compute the hash for cosmwasm contract instatiation params.
+// The hash is keccak256 hash(hash(hash(BigEndian(CodeID)), Label), Msg).
+// We compute the nested hash so there is no chance of bytes leaking between CodeID, Label, and Msg.
```

Comment:
The PR description says this is `sha3.Sum(BigEndian(CodeID) || Label || Msg)`, which doesn't match this comment description.  Also I don't quite understand what "bytes leaking" means in this context.

Reply:
The PR description is wrong - I just updated it.

@kcsongor brought up an interesting point that if `label = AAA` and `msg = BBB`, under the `keccak 256 hash(BigEndian(CodeID) || Label || Msg)`, an attacker could submit an instantiation message with `label = AA` and `msg = ABBB` and it would still hash to the same thing.

It's likely not a major security vulnerability since `msg` is meant to be a json blob, but using the hash of hashes is a bit cleaner and avoids this issue entirely.

Reply:
Ack.  Another option could be to inject a field separator (like "/" or ":") but this is fine as well.